components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpConfiguration.java

```diff
@@ -37,6 +37,8 @@
     private String chmod;
     @UriParam(label = "consumer,advanced")
     private boolean handleDirectoryParserAbsoluteResult;
+    @UriParam(label = "consumer,advanced")
```

Comment:
Set to be producer config option

Reply:
I guess that would be the safest approach, but I don't think trailing slash is ever required.

Since the immediate parent directory creation command never adds a trailing slash,
I just assumed it was a bug that is triggered only on certain file systems that ftp is on.

If trailing slash is required on mkdir command, then immediate parent directory creation will fail on line 292 (of master), so I still don't see much merit on making this configurable.

Reply:
Having said that.  Making it configurable won't hurt anything. I really do not have a firm objection, just a preference.

Reply:
tested with your suggested change too, there are errors. seems safer to go with option.

[ERROR] Failures: 
[ERROR] org.apache.camel.component.file.remote.FileToFtpTempFileNameTest.testFileToFtp(org.apache.camel.component.file.remote.FileToFtpTempFileNameTest)
[ERROR]   Run 1: FileToFtpTempFileNameTest.testFileToFtp:40->Assert.assertTrue:41->Assert.fail:88 File should exists ./target/res/home/out/sub/hello.txt
[ERROR]   Run 2: FileToFtpTempFileNameTest.testFileToFtp:40->Assert.assertTrue:41->Assert.fail:88 File should exists ./target/res/home/out/sub/hello.txt
[ERROR]   Run 3: FileToFtpTempFileNameTest.testFileToFtp:40->Assert.assertTrue:41->Assert.fail:88 File should exists ./target/res/home/out/sub/hello.txt
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FromFtpMoveFileNotStepwiseTest.testPollFileAndShouldBeMoved(org.apache.camel.component.file.remote.FromFtpMoveFileNotStepwiseTest)
[ERROR]   Run 1: FromFtpMoveFileNotStepwiseTest>FromFtpMoveFileTest.testPollFileAndShouldBeMoved:53 mock://result The file should exists: ./target/res/home/movefile/done/sub2/hello.txt.old
[ERROR]   Run 2: FromFtpMoveFileNotStepwiseTest>FromFtpMoveFileTest.testPollFileAndShouldBeMoved:53 mock://result The file should exists: ./target/res/home/movefile/done/sub2/hello.txt.old
[ERROR]   Run 3: FromFtpMoveFileNotStepwiseTest>FromFtpMoveFileTest.testPollFileAndShouldBeMoved:53 mock://result The file should exists: ./target/res/home/movefile/done/sub2/hello.txt.old
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FromFtpMoveFileTest.testPollFileAndShouldBeMoved(org.apache.camel.component.file.remote.FromFtpMoveFileTest)
[ERROR]   Run 1: FromFtpMoveFileTest.testPollFileAndShouldBeMoved:53 mock://result The file should exists: ./target/res/home/movefile/done/sub2/hello.txt.old
[ERROR]   Run 2: FromFtpMoveFileTest.testPollFileAndShouldBeMoved:53 mock://result The file should exists: ./target/res/home/movefile/done/sub2/hello.txt.old
[ERROR]   Run 3: FromFtpMoveFileTest.testPollFileAndShouldBeMoved:53 mock://result The file should exists: ./target/res/home/movefile/done/sub2/hello.txt.old
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpConsumerMoveExpressionTest.testMoveUsingExpression(org.apache.camel.component.file.remote.FtpConsumerMoveExpressionTest)
[ERROR]   Run 1: FtpConsumerMoveExpressionTest.testMoveUsingExpression:66->Assert.assertTrue:41->Assert.fail:88 File should have been renamed
[ERROR]   Run 2: FtpConsumerMoveExpressionTest.testMoveUsingExpression:66->Assert.assertTrue:41->Assert.fail:88 File should have been renamed
[ERROR]   Run 3: FtpConsumerMoveExpressionTest.testMoveUsingExpression:66->Assert.assertTrue:41->Assert.fail:88 File should have been renamed
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerFileWithPathPathSeparatorAutoNoStepwiseTest.testProducerFileWithPathNoStepwise(org.apache.camel.component.file.remote.FtpProducerFileWithPathPathSeparatorAutoNoStepwiseTest)
[ERROR]   Run 1: FtpProducerFileWithPathPathSeparatorAutoNoStepwiseTest.testProducerFileWithPathNoStepwise:43->Assert.assertTrue:41->Assert.fail:88 The uploaded file should exists
[ERROR]   Run 2: FtpProducerFileWithPathPathSeparatorAutoNoStepwiseTest.testProducerFileWithPathNoStepwise:43->Assert.assertTrue:41->Assert.fail:88 The uploaded file should exists
[ERROR]   Run 3: FtpProducerFileWithPathPathSeparatorAutoNoStepwiseTest.testProducerFileWithPathNoStepwise:43->Assert.assertTrue:41->Assert.fail:88 The uploaded file should exists
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerFileWithPathPathSeparatorUnixNoStepwiseTest.testProducerFileWithPathNoStepwise(org.apache.camel.component.file.remote.FtpProducerFileWithPathPathSeparatorUnixNoStepwiseTest)
[ERROR]   Run 1: FtpProducerFileWithPathPathSeparatorUnixNoStepwiseTest.testProducerFileWithPathNoStepwise:43->Assert.assertTrue:41->Assert.fail:88 The uploaded file should exists
[ERROR]   Run 2: FtpProducerFileWithPathPathSeparatorUnixNoStepwiseTest.testProducerFileWithPathNoStepwise:43->Assert.assertTrue:41->Assert.fail:88 The uploaded file should exists
[ERROR]   Run 3: FtpProducerFileWithPathPathSeparatorUnixNoStepwiseTest.testProducerFileWithPathNoStepwise:43->Assert.assertTrue:41->Assert.fail:88 The uploaded file should exists
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerFileWithPathPathSeparatorWindowsNoStepwiseTest.testProducerFileWithPathNoStepwise(org.apache.camel.component.file.remote.FtpProducerFileWithPathPathSeparatorWindowsNoStepwiseTest)
[ERROR]   Run 1: FtpProducerFileWithPathPathSeparatorWindowsNoStepwiseTest.testProducerFileWithPathNoStepwise:43->Assert.assertTrue:41->Assert.fail:88 The uploaded file should exists
[ERROR]   Run 2: FtpProducerFileWithPathPathSeparatorWindowsNoStepwiseTest.testProducerFileWithPathNoStepwise:43->Assert.assertTrue:41->Assert.fail:88 The uploaded file should exists
[ERROR]   Run 3: FtpProducerFileWithPathPathSeparatorWindowsNoStepwiseTest.testProducerFileWithPathNoStepwise:43->Assert.assertTrue:41->Assert.fail:88 The uploaded file should exists
[INFO] 
[ERROR] Errors: 
[ERROR] org.apache.camel.component.file.remote.FromFtpSetNamesWithMultiDirectoriesTest.testFtpRoute(org.apache.camel.component.file.remote.FromFtpSetNamesWithMultiDirectoriesTest)
[ERROR]   Run 1: FromFtpSetNamesWithMultiDirectoriesTest.testFtpRoute:54->prepareFtpServer:86 » GenericFileOperationFailed
[ERROR]   Run 2: FromFtpSetNamesWithMultiDirectoriesTest.testFtpRoute:54->prepareFtpServer:86 » GenericFileOperationFailed
[ERROR]   Run 3: FromFtpSetNamesWithMultiDirectoriesTest.testFtpRoute:54->prepareFtpServer:86 » GenericFileOperationFailed
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpConsumerAutoCreateTest.testAutoCreate(org.apache.camel.component.file.remote.FtpConsumerAutoCreateTest)
[ERROR]   Run 1: FtpConsumerAutoCreateTest.testAutoCreate:46 » GenericFileOperationFailed File ...
[ERROR]   Run 2: FtpConsumerAutoCreateTest.testAutoCreate:46 » GenericFileOperationFailed File ...
[ERROR]   Run 3: FtpConsumerAutoCreateTest.testAutoCreate:46 » GenericFileOperationFailed File ...
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpConsumerRelativeFileNameTest.testValidFilenameOnExchange(org.apache.camel.component.file.remote.FtpConsumerRelativeFileNameTest)
[ERROR]   Run 1: FtpConsumerRelativeFileNameTest.setUp:33->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 2: FtpConsumerRelativeFileNameTest.setUp:33->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 3: FtpConsumerRelativeFileNameTest.setUp:33->FtpServerTestSupport.sendFile:130 » CamelExecution
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerBuildDirectoryTest.testProduceAndBuildFullRemotFolderTest(org.apache.camel.component.file.remote.FtpProducerBuildDirectoryTest)
[ERROR]   Run 1: FtpProducerBuildDirectoryTest.testProduceAndBuildFullRemotFolderTest:35->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 2: FtpProducerBuildDirectoryTest.testProduceAndBuildFullRemotFolderTest:35->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 3: FtpProducerBuildDirectoryTest.testProduceAndBuildFullRemotFolderTest:35->FtpServerTestSupport.sendFile:130 » CamelExecution
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerBuildPartOfDirectoryTest.testProduceAndBuildPartOfRemotFolderTest(org.apache.camel.component.file.remote.FtpProducerBuildPartOfDirectoryTest)
[ERROR]   Run 1: FtpProducerBuildPartOfDirectoryTest.testProduceAndBuildPartOfRemotFolderTest:35->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 2: FtpProducerBuildPartOfDirectoryTest.testProduceAndBuildPartOfRemotFolderTest:35->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 3: FtpProducerBuildPartOfDirectoryTest.testProduceAndBuildPartOfRemotFolderTest:35->FtpServerTestSupport.sendFile:130 » CamelExecution
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerExpressionTest.testProducerComplexByExpression(org.apache.camel.component.file.remote.FtpProducerExpressionTest)
[ERROR]   Run 1: FtpProducerExpressionTest.testProducerComplexByExpression:91 » CamelExecution ...
[ERROR]   Run 2: FtpProducerExpressionTest.testProducerComplexByExpression:91 » CamelExecution ...
[ERROR]   Run 3: FtpProducerExpressionTest.testProducerComplexByExpression:91 » CamelExecution ...
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerFileWithCharsetTest.testProducerWithCharset(org.apache.camel.component.file.remote.FtpProducerFileWithCharsetTest)
[ERROR]   Run 1: FtpProducerFileWithCharsetTest.testProducerWithCharset:54->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 2: FtpProducerFileWithCharsetTest.testProducerWithCharset:54->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 3: FtpProducerFileWithCharsetTest.testProducerWithCharset:54->FtpServerTestSupport.sendFile:130 » CamelExecution
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerFileWithPathNoStepwiseTest.testProducerFileWithPathNoStepwise(org.apache.camel.component.file.remote.FtpProducerFileWithPathNoStepwiseTest)
[ERROR]   Run 1: FtpProducerFileWithPathNoStepwiseTest.testProducerFileWithPathNoStepwise:32->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 2: FtpProducerFileWithPathNoStepwiseTest.testProducerFileWithPathNoStepwise:32->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 3: FtpProducerFileWithPathNoStepwiseTest.testProducerFileWithPathNoStepwise:32->FtpServerTestSupport.sendFile:130 » CamelExecution
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerFileWithPathTest.testProducerFileWithPath(org.apache.camel.component.file.remote.FtpProducerFileWithPathTest)
[ERROR]   Run 1: FtpProducerFileWithPathTest.testProducerFileWithPath:32->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 2: FtpProducerFileWithPathTest.testProducerFileWithPath:32->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 3: FtpProducerFileWithPathTest.testProducerFileWithPath:32->FtpServerTestSupport.sendFile:130 » CamelExecution
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerSendNoopFalseTest.testSendNoopFalse(org.apache.camel.component.file.remote.FtpProducerSendNoopFalseTest)
[ERROR]   Run 1: FtpProducerSendNoopFalseTest.testSendNoopFalse:32->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 2: FtpProducerSendNoopFalseTest.testSendNoopFalse:32->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 3: FtpProducerSendNoopFalseTest.testSendNoopFalse:32->FtpServerTestSupport.sendFile:130 » CamelExecution
[INFO] 
[ERROR] org.apache.camel.component.file.remote.FtpProducerTempPrefixTest.testProduceTempPrefixTest(org.apache.camel.component.file.remote.FtpProducerTempPrefixTest)
[ERROR]   Run 1: FtpProducerTempPrefixTest.testProduceTempPrefixTest:40->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 2: FtpProducerTempPrefixTest.testProduceTempPrefixTest:40->FtpServerTestSupport.sendFile:130 » CamelExecution
[ERROR]   Run 3: FtpProducerTempPrefixTest.testProduceTempPrefixTest:40->FtpServerTestSupport.sendFile:130 » CamelExecution
[INFO] 
[ERROR] org.apache.camel.component.file.remote.RemoteFileProduceOverruleOnlyOnceTest.testFileToFtp(org.apache.camel.component.file.remote.RemoteFileProduceOverruleOnlyOnceTest)
[ERROR]   Run 1: RemoteFileProduceOverruleOnlyOnceTest.testFileToFtp:37 » CamelExecution Except...
[ERROR]   Run 2: RemoteFileProduceOverruleOnlyOnceTest.testFileToFtp:37 » CamelExecution Except...
[ERROR]   Run 3: RemoteFileProduceOverruleOnlyOnceTest.testFileToFtp:37 » CamelExecution Except...
[INFO] 
[ERROR] org.apache.camel.component.file.remote.ToFtpTempFileTargetFileExistTest.testSendFileTargetFileExist(org.apache.camel.component.file.remote.ToFtpTempFileTargetFileExistTest)
[ERROR]   Run 1: ToFtpTempFileTargetFileExistTest.setUp:40->prepareFtpServer:63 » GenericFileOperationFailed
[ERROR]   Run 2: ToFtpTempFileTargetFileExistTest.setUp:40->prepareFtpServer:63 » GenericFileOperationFailed
[ERROR]   Run 3: ToFtpTempFileTargetFileExistTest.setUp:40->prepareFtpServer:63 » GenericFileOperationFailed